src/nrtk_explorer/library/coco_utils.py

```diff
@@ -92,3 +97,51 @@ def convert_from_predictions_to_second_arg(predictions):
         annotations_predictions.append(current_annotations)
 
     return annotations_predictions
+
+
+# Example usage:
+# def is_odd(x):
+#     return x % 2 == 1
+# partition(is_odd, [1, 2, 3, 4])
+def partition(pred, iterable):
+    "Use a predicate to partition entries into false entries and true entries"
+    # partition(is_odd, [1, 2, 3, 4]) --> ([2, 4], [1, 3])
+    it = iter(iterable)
+    true_result, false_result = [], []
+    for elem in it:
+        if pred(elem):
+            true_result.append(elem)
+        else:
+            false_result.append(elem)
+    return true_result, false_result
+
+
+def compute_score(dataset_ids: Sequence[DatasetId], actual, predicted):
+    """Compute score for image ids."""
+
+    # separate images with no predictions in actual argument
+    # as score function expects at least one prediction
+    def is_empty(prediction_pair):
+        actual_predictions = prediction_pair[0]
+        return len(actual_predictions) == 0
+
+    no_predictions, has_predictions = partition(is_empty, zip(actual, predicted, dataset_ids))
+
+    both_images_no_prediction, one_has_prediction = partition(
+        lambda pair: len(pair[1]) == 0, no_predictions
+    )
+    scores = []
+    for _, __, dataset_id in both_images_no_prediction:
+        scores.append((dataset_id, 1.0))
+    for _, __, dataset_id in one_has_prediction:
+        scores.append((dataset_id, 0.0))
+
+    if len(has_predictions) == 0:
+        return scores
+
+    actual, predicted, dataset_ids = zip(*has_predictions)
+    score_output = ClassAgnosticPixelwiseIoUScorer().score(actual, predicted)
```

Comment:
Why moving to` ClassAgnosticPixelWiseIOUScorer()`?

Reply:
Because COCO scorer errored when model outputs category not in the JSON